### PR TITLE
feat: add EventTarget

### PR DESCRIPTION
**Pull request by kettanaito**
## Changes

- Makes this library both Node.js and browser-compatible by adding a `StrictEventTarget` interface. Both Node.js and browser have `EventTarget` implemented, which is fantastic. This will add type-safety on top of those implementations.

### CODE REVIEW

**Comment by kettanaito** [comment]
## Released: v0.3.0 🎉

This has been released in v0.3.0!

- 📄 [**Release notes**](https://github.com/open-draft/strict-event-emitter/releases/tag/v0.3.0)
- 📦 [npm package](https://www.npmjs.com/package/strict-event-emitter/v/0.3.0)

Make sure to always update to the latest version (`npm i strict-event-emitter@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

File: tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,7 @@
 {
   "compilerOptions": {
+    "target": "esnext",
```
Comment (paazmaya) [inline tsconfig.json:3]: A bit late to ask, but wouldnt this make the genrated code to require additional processing?
How about setting it to a released ES version, such as es6?
https://www.typescriptlang.org/tsconfig#target

Comment (kettanaito) [reply tsconfig.json:3]: Yes, you're right! This was an oversight on my part and we need to open a pull request to fix this. I think ES5/ES6 is a sensible target to set instead.

Comment (paazmaya) [reply tsconfig.json:3]: https://github.com/open-draft/strict-event-emitter/pull/13

**Pull request merged**